### PR TITLE
home-manager: remove result in $CWD when doing home-manager build

### PR DESCRIPTION
**Pull request by SuperSandro2000**
### Checklist

- [x] Change is backwards compatible.

- [ ] Code formatted with `./format`.

- [x] Code tested through `nix-shell --pure tests -A run.all`.

- [ ] Test cases updated/added. See [example](https://github.com/nix-community/home-manager/commit/f3fbb50b68df20da47f9b0def5607857fcc0d021#diff-b61a6d542f9036550ba9c401c80f00ef).

- [x] Commit messages are formatted like

    ```
    {component}: {description}

    {long description}
    ```

    See [CONTRIBUTING](https://github.com/nix-community/home-manager/blob/master/doc/contributing.adoc#sec-commit-style) for more information and [recent commit messages](https://github.com/nix-community/home-manager/commits/master) for examples.

- If this PR adds a new module

  - [ ] Added myself as module maintainer. See [example](https://github.com/nix-community/home-manager/blob/068ff76a10e95820f886ac46957edcff4e44621d/modules/programs/lesspipe.nix#L6).

  - [ ] Added myself and the module files to `.github/CODEOWNERS`.

### CODE REVIEW

**Review by berbiche — CHANGES_REQUESTED** [review]
Instead of making this the default, you can add the `--no-out-link` option to the forwarded arguments to `nix-build`: https://github.com/nix-community/home-manager/commit/5eb199e9371b59e97186d83dc4a44477a7c79e43

**Comment by SuperSandro2000** [comment]
Is there any reason you want a stray result link on your CWD when issuing home-manager build? Also this flag cannot be passed to the other nix-build because that would break the build.

**Comment by berbiche** [comment]
> Is there any reason you want a stray result link on your CWD when issuing home-manager build?

I don't use `home-manager` directly nor `home-manager build`.

> Also this flag cannot be passed to the other nix-build because that would break the build.

A new variable can be used for this purpose.
<br>
Anyhow, since I'm not a consumer of this command, I'll leave others to comment on this.

**Comment by rycee** [comment]
Agreed, we definitely cannot change the default behavior here. But adding a `--no-out-link` option for the `build` command seems quite reasonable.

**Comment by SuperSandro2000** [comment]
I have converted --no-out-link into an option and fixed a problem I found along the way.

File: home-manager/home-manager [hunk 1]
```diff
@@ -263,8 +264,9 @@ function doBuild() {
     newsInfo=$(buildNews)
 
     local exitCode
-
-    doBuildAttr --attr activationPackage \
+    doBuildAttr \
+                ${NO_OUT_LINK+--no-out-link} \
+                --attr activationPackage \
```
Comment (SuperSandro2000) [inline home-manager/home-manager:269]:
```suggestion
    doBuildAttr ${NO_OUT_LINK+--no-out-link} \
                --attr activationPackage \
```

File: home-manager/home-manager [hunk 2]
```diff
@@ -263,8 +264,9 @@ function doBuild() {
     newsInfo=$(buildNews)
```
Comment (SuperSandro2000) [inline home-manager/home-manager:264]: The if here is required because the script tries to load the news file which will fail and the error is annoying.

Comment (rycee) [reply home-manager/home-manager:264]: Hmm, I don't get any errors. Can you paste the error?

File: home-manager/home-manager (hunk 1 above)
Comment (rycee) [reply home-manager/home-manager:269]: I did it this way to match the other calls to `doBuildAttr`. If the style should change then better do all of them together in one commit.

File: home-manager/home-manager (hunk 2 above)
Comment (SuperSandro2000) [reply home-manager/home-manager:264]:
``` 
+ /tmp/home-manager-build.xLOtx8D8bR/generation/activate
$HOME/.nix-profile/bin/home-manager: line 311: /tmp/home-manager-build.xLOtx8D8bR/generation/activate: No such file or directory
+ exitCode=1
```

**Pull request merged**